Parser/asdl_c.py

```diff
@@ -163,6 +163,44 @@ def visitProduct(self, product, name, depth):
         self.emit(s, depth)
         self.emit("", depth)
 
+class SequenceDefVisitor(EmitVisitor):
+    def visitModule(self, mod):
+        for dfn in mod.dfns:
+            self.visit(dfn)
+
+    def visitType(self, type, depth=0):
+        self.visit(type.value, type.name, depth)
+
+    def visitSum(self, sum, name, depth):
+        if not is_simple(sum):
+            self.sum_with_constructors(sum, name, depth)
+
+    def emit_sequence_constructor(self, name, type, depth):
+        self.emit("asdl_%(name)s_seq *_Py_asdl_%(name)s_seq_new(Py_ssize_t size, PyArena *arena);" % locals(), depth)
+
+    def sum_with_constructors(self, sum, name, depth):
+        ctype = get_c_type(name)
+        s = """
+typedef struct {
+    _ASDL_SEQ_HEAD
+    %(ctype)s typed_elements[1];
+} asdl_%(name)s_seq;""" % locals()
+        self.emit(s, reflow=False, depth=depth)
+        self.emit("", depth)
+        self.emit_sequence_constructor(name, ctype, depth)
+        self.emit("", depth)
```

Comment:
Would it be feasible to merge the implementation of complex sums with the products since they are the same right now?

Reply:
Done in 2d06901